chasm/search_attribute.go

```diff
@@ -382,3 +384,166 @@ func (s SearchAttributeKeywordList) Value(value []string) SearchAttributeKeyValu
 		Value: VisibilityValueStringSlice(value),
 	}
 }
+
+// SearchAttributeMap wraps search attribute values with type-safe access.
+type SearchAttributesMap struct {
+	values map[string]VisibilityValue
+}
+
+// NewSearchAttributeMap creates a new SearchAttributeMap from raw values.
+func NewSearchAttributesMap(values map[string]VisibilityValue) SearchAttributesMap {
+	return SearchAttributesMap{values: values}
+}
+
+// GetBool returns the boolean value for a given SearchAttributeBool. If not found or map is nil, second parameter is false.
+func (m SearchAttributesMap) GetBool(sa SearchAttributeBool) (bool, bool) {
```

Comment:
I think we can probably use generics here and have just a single `Get` method.
It would require defining another `TypedSearchAttribute[T]` interface with a single `typeMarker(T)` method and making sure that `SearchAttributeBool` and friend implement that interface.